mantidproject/mantid
# Add peak picker to ALFView analysis plot

**Description of work.**
This PR adds a peak picker tool to the PreviewPlot in the analysis part of the ALFView interface. This makes it easier to adjust the peak centre in a visual way.

**To test:**
Open ALFView and load ALF82301
Click on Pick tab
Expand Rebin section. Rebin with 5.5,0.01,6
Click the Rectangle region selection tool at the top of the Pick tab
Select the two tubes in the centre with the yellow peak
The peak centre estimate should be updated automatically. There should be a grey line roughly down the centre of the peak,
Try and move this grey line. It will turn red when you grab it, and moving it should cause the peak centre box to update
clicking on the fit or update estimate button should deselect the peak picker.

Fixes #34604

*This does not require release notes* because **release notes will be added in a later PR**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
